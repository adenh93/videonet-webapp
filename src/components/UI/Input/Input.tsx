import styled from "styled-components";
import { deviceWidths } from "../../../CONSTANTS";

const Input = styled.input`
  color: ${({ theme }) => theme.palette.text.input};
  font-size: ${({ theme }) => theme.fontSize.input};
  font-weight: ${({ theme }) => theme.fontWeights.light};
  background-color: ${({ theme }) => theme.palette.background};
  font-family: "Lato", sans-serif;
  letter-spacing: 0.03em;
  height: 6rem;
  width: 73.5rem;
  border-radius: ${({ theme }) => theme.borderRadius.rounded};
  padding: 1rem 14rem 1rem 8rem;
  border: 0;
  filter: drop-shadow(${({ theme }) => theme.shadows.med});

  @media (max-width: ${deviceWidths.tablet}) {
    width: 100%;
  }
`;

export default Input;
