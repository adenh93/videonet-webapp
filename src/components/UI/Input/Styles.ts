import styled from "styled-components";

export const InputContainer = styled.span`
  position: relative;
`;

export const StyledInput = styled.input`
  color: ${({ theme }) => theme.palette.text.input};
  font-size: ${({ theme }) => theme.fontSize.input};
  font-weight: ${({ theme }) => theme.fontWeights.light};
  background-color: ${({ theme }) => theme.palette.background};
  font-family: "Lato", sans-serif;
  letter-spacing: 0.03em;
  height: 6rem;
  width: 73.5rem;
  border-radius: 20rem;
  padding: 1rem 12rem 1rem 8rem;
  border: 0;
  filter: drop-shadow(${({ theme }) => theme.shadows.med});
`;

export const InputIcon = styled.img`
  width: 2.6rem;
  position: absolute;
  z-index: 3;
  transform: translate(130%, 65%);
`;
