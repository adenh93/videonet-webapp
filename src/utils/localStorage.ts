import { FilmDetails, SearchResult } from "../graphql/types";

type LocalStorageCollection = "WATCHED_ITEMS" | "LIKED_ITEMS";

export const setItem = <T>(type: LocalStorageCollection, item: T): void =>
  localStorage.setItem(type, JSON.stringify(item));

export const getItems = <R>(type: LocalStorageCollection): R[] => {
  const items = localStorage.getItem(type);
  if (!items) return [];
  return JSON.parse(items) as R[];
};

export const getWatchList = (): SearchResult[] =>
  getItems<SearchResult>("WATCHED_ITEMS");

export const isInWatchList = (movieId: number): boolean =>
  getWatchList().find((m) => m.id === movieId) !== undefined;

export const addToWatchList = (movie: FilmDetails): void => {
  const movies = getWatchList();
  if (!movies.find((m) => m.id === movie.id))
    setItem("WATCHED_ITEMS", [...movies, movie]);
};

export const removeFromWatchList = (movieId: number): void =>
  setItem(
    "WATCHED_ITEMS",
    getWatchList().filter((m) => m.id !== movieId)
  );

export const getLikedMovies = (): number[] => getItems<number>("LIKED_ITEMS");

export const isLiked = (movieId: number): boolean =>
  getLikedMovies().find((item) => item === movieId) !== undefined;

export const likeMovie = (movieId: number): void => {
  const likedItems = getLikedMovies();
  if (!likedItems.find((item) => item === movieId))
    setItem("LIKED_ITEMS", [...likedItems, movieId]);
};

export const unlikeMovie = (movieId: number): void => {
  setItem(
    "LIKED_ITEMS",
    getLikedMovies().filter((item) => item !== movieId)
  );
};

export const toggleWatched = (movie: FilmDetails): boolean => {
  const watched = isInWatchList(movie.id);
  watched ? removeFromWatchList(movie.id) : addToWatchList(movie);
  return !watched;
};

export const toggleLiked = (movieId: number): boolean => {
  const liked = isLiked(movieId);
  liked ? unlikeMovie(movieId) : likeMovie(movieId);
  return !liked;
};
