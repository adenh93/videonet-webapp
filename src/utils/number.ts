import numeral from "numeral";

export const formatNumber = (n: Number): String => numeral(n).format("0,0");
