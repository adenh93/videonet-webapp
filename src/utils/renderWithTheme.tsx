import React, { ReactElement } from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { BrowserRouter as Router } from "react-router-dom";
import theme from "../theme";

export const renderWithTheme = (component: ReactElement) =>
  renderer.create(
    <ThemeProvider theme={theme}>
      <Router>{component}</Router>
    </ThemeProvider>
  );

export const rtlWithTheme = (component: ReactElement) =>
  render(
    <ThemeProvider theme={theme}>
      <Router>{component}</Router>
    </ThemeProvider>
  );
