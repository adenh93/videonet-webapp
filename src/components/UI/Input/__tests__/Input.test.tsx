import React from "react";
import { renderWithTheme } from "../../../../utils/renderWithTheme";
import Input from "../Input";
import "jest-styled-components";

describe("Input UI component", () => {
  it("renders correctly", () => {
    const tree = renderWithTheme(<Input>Test</Input>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
