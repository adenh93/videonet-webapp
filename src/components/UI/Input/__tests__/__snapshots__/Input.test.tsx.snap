// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input UI component renders correctly 1`] = `
.c0 {
  color: #333333;
  font-size: 1.8rem;
  font-weight: 300;
  background-color: #f9f9f9;
  font-family: "Lato",sans-serif;
  -webkit-letter-spacing: 0.03em;
  -moz-letter-spacing: 0.03em;
  -ms-letter-spacing: 0.03em;
  letter-spacing: 0.03em;
  height: 6rem;
  width: 73.5rem;
  border-radius: 20rem;
  padding: 1rem 14rem 1rem 8rem;
  border: 0;
  -webkit-filter: drop-shadow(0 0.4rem 0.4rem rgba(0,0,0,0.25));
  filter: drop-shadow(0 0.4rem 0.4rem rgba(0,0,0,0.25));
}

<input
  className="c0"
>
  Test
</input>
`;
